src/main/java/controller/BusController.java

```diff
@@ -1,28 +1,94 @@
 package controller;
 
 import db.DatabaseHandler;
-
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import entity.Bus;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 
 public class BusController {
-	DatabaseHandler databaseHandler = new DatabaseHandler();
-	Statement statement = databaseHandler.createStatement();
 
-	public void listAllBuses() {
+	public int listAllBuses() {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+		int id = -1;
+		System.out.println("List of all buses:");
 		try {
+			if (databaseHandler.getDbConnection() != null) {
 			ResultSet resultSetBuses = statement.executeQuery("SELECT * FROM buses");
 			while (resultSetBuses.next()) {
-				int myDriverId = resultSetBuses.getInt("id");
-				String myBusNumber = resultSetBuses.getString("busNumber");
-				float myBusFuel = resultSetBuses.getFloat("fuel");
-				System.out.println(myDriverId + " " + myBusNumber + " " + myBusFuel);
+				id = resultSetBuses.getInt("id");
+				String busNumber = resultSetBuses.getString("busNumber");
+				int driverId = resultSetBuses.getInt("driverId");
+				float fuel = resultSetBuses.getFloat("fuel");
+				Timestamp createdOn = resultSetBuses.getTimestamp("createdOn");
+				Date purchasedOn = resultSetBuses.getDate("purchasedOn");
+
+				Bus newBus = new Bus(id, busNumber, driverId, fuel, createdOn, purchasedOn);
+
+				System.out.println(newBus.toString());
 			}
+			statement.close();
 			databaseHandler.closeConnection();
-
+			}
+			return id;
 		} catch (SQLException ex) {
-			System.out.println(ex);
+			ex.printStackTrace();
+			return id;
+		}
+	}
+
+	public int findBusByNumber(String busNumber) {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+		int id = -1;
+		try {
+			if (databaseHandler.getDbConnection() != null) {
+			ResultSet resultSetBuses = statement.executeQuery(
+					"SELECT * FROM buses where busNumber = '" + busNumber + "' limit 1");
+			boolean next = resultSetBuses.next();
+				if (next) {
+					id = resultSetBuses.getInt("id");
+					busNumber = resultSetBuses.getString("busNumber");
+					int driverId = resultSetBuses.getInt("driverId");
+					float fuel = resultSetBuses.getFloat("fuel");
+					Timestamp createdOn = resultSetBuses.getTimestamp("createdOn");
+					Date purchasedOn = resultSetBuses.getDate("purchasedOn");
+					Bus newBus = new Bus(id, busNumber, driverId, fuel, createdOn, purchasedOn);
+
+					System.out.println("Bus with id: " + id);
+					System.out.println(newBus.toString());
+				}
+			}
+			return id;
```

Comment:
are you returning -1 if there is no connection to DB?
probably more meaning text or error message would make more sense?

And what happens if there are no buses found? Do we ignore search?

Reply:
It is better to return integer, because you could use this in other parts of code for example for some checking. For visual purposes we have **Exception** message instead.

If no bus would be found, an exception will be thrown

Reply:
good to know, makes sense